x/tradeshield/keeper/msg_server_perpetual_order.go

```diff
@@ -100,47 +73,51 @@ func (k msgServer) CreatePerpetualOpenOrder(goCtx context.Context, msg *types.Ms
 		return nil, err
 	}
 
+	// emit event for limit open order created
+	ctx.EventManager().EmitEvent(types.NewCreatePerpetualOpenOrderEvt(pendingPerpetualOrder))
+
 	return &types.MsgCreatePerpetualOpenOrderResponse{
 		OrderId: pendingPerpetualOrder.OrderId,
 	}, nil
 }
 
 func (k msgServer) CreatePerpetualCloseOrder(goCtx context.Context, msg *types.MsgCreatePerpetualCloseOrder) (*types.MsgCreatePerpetualCloseOrderResponse, error) {
-	// Disable for v1
-	return nil, errorsmod.Wrap(sdkerrors.ErrInvalidRequest, "disabled for v1")
-	// ctx := sdk.UnwrapSDKContext(goCtx)
-
-	// // check if the position owner address matches the msg owner address
-	// position, err := k.perpetual.GetMTP(ctx, sdk.MustAccAddressFromBech32(msg.OwnerAddress), msg.PositionId)
-	// if err != nil {
-	// 	return nil, errorsmod.Wrap(sdkerrors.ErrKeyNotFound, fmt.Sprintf("position %d not found", msg.PositionId))
-	// }
-
-	// var pendingPerpetualOrder = types.PerpetualOrder{
-	// 	PerpetualOrderType: types.PerpetualOrderType_LIMITCLOSE,
-	// 	TriggerPrice: types.TriggerPrice{
-	// 		TradingAssetDenom: position.TradingAsset,
-	// 		Rate:              msg.TriggerPrice.Rate,
-	// 	},
-	// 	OwnerAddress: position.Address,
-	// 	PositionId:   position.Id,
-	// }
-
-	// id := k.AppendPendingPerpetualOrder(
-	// 	ctx,
-	// 	pendingPerpetualOrder,
-	// )
-
-	// return &types.MsgCreatePerpetualCloseOrderResponse{
-	// 	OrderId: id,
-	// }, nil
+	ctx := sdk.UnwrapSDKContext(goCtx)
+
+	// check if the position owner address matches the msg owner address
+	position, err := k.perpetual.GetMTP(ctx, msg.PoolId, sdk.MustAccAddressFromBech32(msg.OwnerAddress), msg.PositionId)
+	if err != nil {
+		return nil, errorsmod.Wrap(sdkerrors.ErrKeyNotFound, fmt.Sprintf("position %d not found", msg.PositionId))
+	}
+
+	var pendingPerpetualOrder = types.PerpetualOrder{
+		PerpetualOrderType: types.PerpetualOrderType_LIMITCLOSE,
+		TriggerPrice:       msg.TriggerPrice,
+		OwnerAddress:       position.Address,
+		PositionId:         position.Id,
+		PoolId:             msg.PoolId,
+		Status:             types.Status_PENDING,
+		ClosePercentage:    msg.ClosePercentage,
+	}
+
+	id := k.AppendPendingPerpetualOrder(
+		ctx,
+		pendingPerpetualOrder,
+	)
+
+	// emit event for limit close order created
+	ctx.EventManager().EmitEvent(types.NewCreatePerpetualCloseOrderEvt(pendingPerpetualOrder))
```

Comment:
this event does not have value id right?

Reply:
order id ?